rdkcentral/Dobby
# RDKTV-12717/RDKTV-13079: Support relaying syslog/journald messages

### Description
When containers attempt to log directly to journald (e.g. `sd_journal_print()`) or syslog, the log messages may not correctly appear in journald.

After adding `/run/systemd/journal/socket` into a container that directly calls `sd_journal_print`, the log messages are seen in `journalctl -f`. However, these messages are not tagged with a `_SYSTEMD_UNIT` tag. This is because the message unit is determined using the PID of the process, which does not match with a known unit PID when the process runs in a container.

As a result, when filtering the logs by unit (`journalctl -fu dobby`), the messages do not appear. This is an issue with RDK log collection scripts.

This PR makes the following changes:
* Based on the relevant setting in `/etc/dobby.json`, create dobby syslog and journald sockets that can be mounted into a container instead. When messages are received on these sockets, Dobby will forward the messages to the "real" socket on the host
    * Since Dobby sends the message to journald, the PID resolves to DobbyDaemon and the messages can be seen when doing `journalctl -fu dobby`
* Re-write much of Dobby logging code to use an epoll based implementation instead of spinning up a thread per-container
    * Since log relays are using epoll, it made sense to standardise all the logging methods to epoll at the same time
    * Improves performance and helps prevent lock-up during shutdown whilst we wait for loggers to finish
    * Observed significantly reduced CPU usage from DobbyDaemon during periods of high log output

### Test Procedure
**This is a high-risk change due to amount of code changed - please test thoroughly**

Create a simple C program with the following contents

```c
#include <systemd/sd-journal.h>
#include <syslog.h>

#include <stdio.h>
#include <unistd.h>

int main(int argc, char const *argv[])
{
    for (;;)
    {
        fprintf(stdout, "Hello world - stdout test!\n");
        fprintf(stderr, "Hello world - stderr test\n");
        syslog(LOG_INFO, "Hello world, this is a syslog test");
        sd_journal_print(LOG_INFO, "Hello world, this is a journald test");

        sleep(1);
    }

    return 0;
}
```

Create a container config to run this program.

Do the following (in Vagrant):
* Start Dobby as systemd service
* Add `/dev/log` and `/run/systemd/journal/socket` mounts to the container. 
    * Run `journalctl -f`. Observe the syslog and journald messages
    * Run `journalctl -f -u dobby`. Observe the syslog and journald messages **do not appear**

* Remove previous bind-mounts. Add `/tmp/dobby-syslog` and `/tmp/dobby-journald` mounts to the container. 
    * Run `journalctl -f`. Observe the syslog and journald messages
    * Run `journalctl -f -u dobby`. Observe the syslog and journald messages **do appear**

Due to the changes elsewhere to logging code, also test the following features:
* Logging to file works correctly
    * Ensure logs are correctly written even if container fails to start
    * Ensure log files are correctly re-written each time container starts (instead of appending)
* Logging to journald correctly
    * When journald logging enabled with container, with the test code provided earlier, observe all 4 messages appearing in journald
* Logging to devnull sink works correctly (e.g. no logs printed)
* Ensure Dobby shuts down without error

### Type of Change

### Requires Bitbake Recipe changes?
